sdk/azidentity/managed_identity_client.go

```diff
@@ -129,10 +139,10 @@ func (c *managedIdentityClient) sendAuthRequest(ctx context.Context, msiType msi
 func (c *managedIdentityClient) createAccessToken(res *azcore.Response) (*azcore.AccessToken, error) {
 	value := struct {
 		// these are the only fields that we use
-		Token        string      `json:"access_token"`
-		RefreshToken string      `json:"refresh_token"`
-		ExpiresIn    json.Number `json:"expires_in"` // this field should always return the number of seconds for which a token is valid
-		ExpiresOn    string      `json:"expires_on"` // the value returned in this field varies between a number and a date string
+		Token        string         `json:"access_token,omitempty"`
+		RefreshToken string         `json:"refresh_token,omitempty"`
+		ExpiresIn    *wrappedNumber `json:"expires_in,omitempty"` // this field should always return the number of seconds for which a token is valid
```

Comment:
I'm guessing that now with `wrappedNumber` this no longer needs to be a pointer?